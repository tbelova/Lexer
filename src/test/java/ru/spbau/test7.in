if (a >= 5 * 185 && x > 10 - 7 || b == 9 * 14 + 13 - 5 * 10 && c <= -1) 
then { 
	return 0; 
}
