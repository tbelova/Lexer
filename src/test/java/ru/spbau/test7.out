PROGRAM(1,44)
   STATEMENTS(1,43)
      STATEMENT(1,43)
         IF_STATEMENT(1,43)
            BOOLEAN_EXPRESSION(3,36)
               OR(3,36)
                  AND(3,16)
                     GE(3,9)
                        EXPRESSION(3,3)
                           IDENT(3,3)[a]
                        EXPRESSION(5,9)
                           MULT(5,9)
                              INTEGER(5,5)[5]
                              INTEGER(7,9)[185]
                     G(11,16)
                        EXPRESSION(11,11)
                           IDENT(11,11)[x]
                        EXPRESSION(13,16)
                           SUB(13,16)
                              INTEGER(13,14)[10]
                              INTEGER(16,16)[7]
                  AND(18,36)
                     EQ(18,31)
                        EXPRESSION(18,18)
                           IDENT(18,18)[b]
                        EXPRESSION(20,31)
                           SUB(20,31)
                              SUM(20,26)
                                 MULT(20,23)
                                    INTEGER(20,20)[9]
                                    INTEGER(22,23)[14]
                                 INTEGER(25,26)[13]
                              MULT(28,31)
                                 INTEGER(28,28)[5]
                                 INTEGER(30,31)[10]
                     LE(33,36)
                        EXPRESSION(33,33)
                           IDENT(33,33)[c]
                        EXPRESSION(35,36)
                           INTEGER(36,36)[1]
            THEN_STATEMENT(39,43)
               BLOCK(39,43)
                  STATEMENTS(40,42)
                     STATEMENT(40,42)
                        RETURN(40,41)
                           EXPRESSION(41,41)
                              INTEGER(41,41)[0]
                     STATEMENTS(43,42)
      STATEMENTS(44,43)
